packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -461,7 +461,7 @@ Future<T> showModalBottomSheet<T>({
     theme: Theme.of(context, shadowThemeOnly: true),
     isScrollControlled: isScrollControlled,
     barrierLabel: MaterialLocalizations.of(context).modalBarrierDismissLabel,
-    backgroundColor: backgroundColor,
+    backgroundColor: backgroundColor ?? Theme.of(context).bottomSheetTheme.modalBackgroundColor,
```

Comment:
Shouldn't this be: 
```
 backgroundColor ?? Theme.of(context).bottomSheetTheme.modalBackgroundColor ?? Theme.of(context).bottomSheetTheme.backgroundColor;
```
It would be best to allow  `Theme.of(context).bottomSheetTheme` to be null and to cache the value, so:
```
final BottomSheetThemeData theme = Theme.of(context).bottomSheetTheme;

backgroundColor: backgroundColor ?? theme?.modalBackgroundColor ?? theme?.backgroundColor;
``

Reply:
Done